core/src/wsv.rs

```diff
@@ -625,20 +651,224 @@ impl<W: WorldTrait> WorldStateView<W> {
         transactions.sort();
         transactions
     }
+
+    /// Register `elem` on behalf of `authority`.
+    ///
+    /// # Errors
+    /// Any error during `elem.register()` call
+    pub fn register<T: RegisterTrait<W>>(
+        &self,
+        elem: T,
+        authority: AccountId,
+    ) -> Result<(), Error> {
+        elem.register(authority, self)
+    }
+
+    /// Unregister object by `id`
+    ///
+    /// # Errors
+    /// Any error during `T::unregister()` call
+    pub fn unregister<T: RegisterTrait<W>>(&self, id: T::Id) -> Result<(), Error> {
+        T::unregister(id, self)
+    }
 }
 
-impl Deref for World {
-    type Target = Self;
-    #[inline]
-    fn deref(&self) -> &Self::Target {
-        self
+/// Trait for objects that can be registered and unregistered
+pub trait RegisterTrait<W: WorldTrait>: Identifiable {
```

Comment:
This could be replaced with a marker trait. Not everything that can be registered should necessarily also be unregistered.